node/node.go

```diff
@@ -130,6 +133,82 @@ func StateProvider(stateProvider statesync.StateProvider) Option {
 	}
 }
 
+// This function is expected to synchronize CometBFTs
+// stores with the application after statesync has been performed offline
+// It is expected that the blockstore and statestore are empty at
+// the time the function is called
+func BootstrapState(height uint64, appHash []byte, config *cfg.Config) error {
+	ctx := context.Background()
+	logger := log.NewTMLogger(log.NewSyncWriter(os.Stdout))
+	if config == nil {
+		config = cfg.DefaultConfig()
+	}
+	blockStore, stateDB, err := initDBs(config, cfg.DefaultDBProvider)
+
+	if err != nil {
+		return err
+	}
+
+	stateStore := sm.NewStore(stateDB, sm.StoreOptions{
+		DiscardABCIResponses: config.Storage.DiscardABCIResponses,
+	})
+
+	if !blockStore.IsEmpty() {
+		logger.Error("overwriting non empty block store")
```

Comment:
Can we bail out instead? and have the operator delete it (at their own risk)?

Reply:
Also, can we do the same check with the state store?

Reply:
I added the check to confimr that the state is indeed empty.